sympy/solvers/tests/test_diophantine.py

```diff
@@ -583,6 +582,58 @@ def check_solutions(eq):
 
     return okay
 
+
+def check_linear(eq, param=symbols("t", Integer=True)):
```

Comment:
The purpose of the test is to catch regressions before they are merged to master.This check does in no way indicate if a solution was reduced to a partial solution after some modification. It will only show that the remaining values do satisfy the equation. 

> The original tests only checked that the output matched the test case. 

I don't think this would necessarily be wrong. A change of output will be quickly revealed.

The result should be verified when the test is written. It should not be necessary to repeatedly verify the correctness of the original result. (The new tests take twice the time of the old test_linear.)